lib/grr.js

```diff
@@ -1399,13 +1403,244 @@ function stepListCRs(arg, cb) {
             if (subject.length > SUBJ_LIMIT) {
                 subject = subject.slice(0, SUBJ_LIMIT) + '...';
             }
+            // Some test gerrit instances don't have emails set for
+            // all users.
+            if (gerrit_res.owner['email'] === null) {
+                gerrit_res.owner['email'] = '<unknown>';
+            }
             console.log(
                 extsprintf(
                     fmt, created.toISOString(), gerrit_res.url,
                     gerrit_res.owner['email'], subject));
         }
     });
 }
+
+/*
+ * Search for open code reviews for *any* repository that match our issue
+ */
+function stepQueryCRs(arg, cb) {
+    assert.object(arg.log, 'arg.log');
+    assert.optionalBool(arg.dryRun, 'arg.dryRun');
+    assert.optionalBool(arg.verbose, 'arg.verbose');
+    assert.string(arg.issueArg, 'arg.issueArg');
+    assert.string(arg.gerritUsername, 'arg.gerritUsername');
+    assert.object(arg.grrConfig, 'arg.grrConfig');
+    assert.func(cb, 'cb');
+
+    arg.grrReviews = [];
+    common.forkExecWaitAndLog({
+        argv: ['ssh', '-q', '-o', 'StrictHostKeyChecking=no',
+            '-o', 'UserKnownHostsFile=/dev/null', '-p', '29418',
+            arg.gerritUsername + '@' + GERRIT_HOST,
+            'gerrit', 'query', '--format=JSON',
+            '--all-approvals', '--current-patch-set',
+            'status:open',
+            'intopic:' + arg.issueArg, 'OR', 'message:' + arg.issueArg],
+        log: arg.log
+    }, function (err, res) {
+        if (err) {
+            cb(err);
+            return;
+        }
+        var lines = res.stdout.trim().split(/\n/g);
+        if (lines.length == 1) {
+            console.log(
+                'No outstanding code reviews matching ' + arg.issueArg + '.');
+            return;
+        }
+        for (var i = 0; i < lines.length; i++) {
+            var element = lines[i];
+            var cr_res = JSON.parse(element);
+            // the last bit of js from gerrit is a row-count + status which
+            // has an empty url field.
+            if (!cr_res.hasOwnProperty('url')) {
+                continue;
+            } else if (!cr_res.open) {
+                // despite the status:open query, our gerrit server returns
+                // open:false results when --all-approvals is passed as a
+                // query parameter.
+                continue;
+            } else {
+                arg.grrReviews.push(cr_res);
+            }
+        }
+        cb();
+    });
+}
+
+/*
+ * List reviews for those CRs
```

Comment:
Perhaps "Work through each of the given `grrReviews`" ?

Reply:
That's better, thanks.